search/handlers.go

```diff
@@ -83,10 +105,89 @@ func (s *Server) handleSearchRequestProfiles(e echo.Context) error {
 		})
 	}
 
-	out, err := s.SearchProfiles(ctx, q)
+	offset, limit, err := parseCursorLimit(e)
+	if err != nil {
+		return err
+	}
+
+	typeahead := false
+	if q := strings.TrimSpace(e.QueryParam("typeahead")); q == "true" || q == "1" || q == "y" {
+		typeahead = true
+	}
+
+	out, err := s.SearchProfiles(ctx, q, typeahead, offset, limit)
 	if err != nil {
 		return err
 	}
 
 	return e.JSON(200, out)
 }
+
+func (s *Server) SearchPosts(ctx context.Context, q string, offset, size int) (*SearchPostsSkeletonResp, error) {
+	resp, err := DoSearchPosts(ctx, s.dir, s.escli, s.postIndex, q, offset, size)
+	if err != nil {
+		return nil, err
+	}
+
+	posts := []syntax.ATURI{}
+	for _, r := range resp.Hits.Hits {
+		var doc PostDoc
+		if err := json.Unmarshal(r.Source, &doc); err != nil {
+			return nil, fmt.Errorf("decoding post doc from search response: %w", err)
```

Comment:
Do we want this to fail the entire query or just log it with some context and move to the next result?

Reply:
This is a pretty bad error which should "never" happen, so I think we want to fail-and-bail on this situation.

An example where this would happen is if there was a config mistake where a query worker got pointed at the wrong elasticsearch index (either post instead of profile, or an older schema version). I think in that kind of situation we want to fail loudly.